src/vos/README.md

```diff
@@ -448,36 +450,55 @@ require such.  Conditional operations are implemented using a combination of
 existence checks and read timestamps.   The read timestamps enable limited
 MVCC to prevent read/write races and provide atomicity guarantees.
 
-<a id="821"></a>
+<a id="821"><a>
+### VOS Timestamp Cache
+
+VOS maintains an in-memory cache of read and write timestamps in order to
+enforce MVCC semantics.  The timestamp cache itself consists of two parts:
+
+1. A global array per target for each type of entity including objects, dkeys,
+and akeys.  The index at each level is determined by the combination of the
+index of the parent entity, or 0 in the case of containers, and the hash
+of the entity in question.   If two different keys map to the same index,
+they share timestamp entries.   This will result in some false conflicts
+but does not affect correctness so long as progress can be made.  The purpose
+of this array is to store timestamps for entries that do not exist in the
```

Comment:
Might worth considering putting this purpose up front, at the beginning of this paragraph. For instance,

    1. <DATA STRUCTURE NAME HINTING THE PURPOSE>. A global array per ...

or

    1. A global array per target that stores timestamps for entries ...